src/services/services.ts

```diff
@@ -3345,6 +3347,20 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName);
+        const imports: Set<string> = new Set();
+        let specifiers = file.imports;
+        specifiers = specifiers.slice(specifiers.findIndex(nodeIsSynthesized) + 1);
+        if (specifiers.length === 0) return [];
+        for (const specifier of specifiers) {
+            const name = program.getResolvedModuleFromModuleSpecifier(specifier, file)?.resolvedModule?.resolvedFileName;

```

Comment:
it may happen that we have resolvedFileName = say "module.js" but its not part of program. Is this ok to send back that file name ?

Reply:
Yes, files that are not part of the program are fine; we don't expect to get information from Typescript about them -- they are fine if they exist on disk.